ansible/undercloud/deploy-undercloud.yml

```diff
@@ -215,16 +215,43 @@
         owner: stack
         group: stack
 
-    - name: check if credentials in instackenv.json are correct
+    - name: Newton(10)/Ocata(11)/Pike(12) check if credentials in instackenv.json are correct
       shell: openstack baremetal instackenv validate  -f  /home/stack/instackenv.json
       register: node_status
       failed_when: "'SUCCESS' not in node_status.stdout"
-
-    - name: Import instackenv.json
-      shell: . /home/stack/stackrc; openstack baremetal import --json /home/stack/instackenv.json
-
-    - name: Configure boot
-      shell: . /home/stack/stackrc; openstack baremetal configure boot
+      when: version < 13
+
+    - name: Queens(13) Import instackenv.json
+      shell: |
+        . /home/stack/stackrc
+        openstack overcloud node import /home/stack/instackenv.json
+      when: version >= 13
+
+    - name: Newton(10)/Ocata(11)/Pike(12) Import instackenv.json
+      shell: |
+        . /home/stack/stackrc
+        openstack baremetal import --json /home/stack/instackenv.json
+      when: version < 13
+
+    - name: Queens(13) Configure boot
+      shell: |
+        . /home/stack/stackrc
+        openstack overcloud node configure --all-manageable
+      when: version >= 13
+
+    - name: Newton(10)/Ocata(11)/Pike(12) Configure boot
+      shell: |
+        . /home/stack/stackrc
+        openstack baremetal configure boot
+      when: version < 13
+
+    - name: Queens(13) Set Nodes to provide
+      shell: |
+        . /home/stack/stackrc
+        { time openstack overcloud node provide --all-manageable 2>&1 | tee -a /home/stack/log/005-overcloud-node-provide.log ; } 2>> /home/stack/log/005-overcloud-node-provide.log
```

Comment:
Does log directory exist at this point? Not entirely sure why u are teeing and also redirecting here?

Reply:
Holy cut and paste error.  Yeah my bad I will patch up to fix this.  

In Microcloud environment I tee output into a separate log directory to capture timings and log output but we don't do that same in this playbook.